File: go/260. Single Number III.go
func singleNumber(nums []int) []int {
    total := 0
    for _, i := range nums {
        total ^= i
    }
    set := 0
    for ; set < 32; set++ {
        if total & (1 << set) > 0 {
            break
        }
    }
    answer1 := 0
    answer2 := 0
    for _, i := range nums {
        if i & (1 << set) > 0 {
            answer1 ^= i
        } else {
            answer2 ^= i
        }
    }
    return []int{answer1, answer2}
}

File: python/260. Single Number III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        total = 0
        for i in nums:
            total ^= i
        set = 0
        while set < 32:
            if total & (1 << set) > 0:
                break
            set += 1
        answer1 = 0
        answer2 = 0
        for i in nums:
            if i & (1 << set) > 0:
                answer1 ^= i
            else:
                answer2 ^= i
        return [answer1, answer2]
